packages/neo-one-client-core/src/user/DapiUserAccountProvider.ts

```diff
@@ -0,0 +1,618 @@
+import BN from 'bn.js';
+import { BehaviorSubject, Observable } from 'rxjs';
+import { distinctUntilChanged } from 'rxjs/operators';
+import { clientUtils } from '../clientUtils';
+import { UnknownProviderError } from '../errors';
+import { Hash256 } from '../Hash256';
+import { Argument, AssetInput, AssetOutput, Dapi, DapiNetworks } from './Dapi';
+import { FullTransfer, Provider, UserAccountProviderBase } from './UserAccountProviderBase';
+
+const paramToArgDataType = (value: ScriptBuilderParam | undefined): Argument => {
+  if (value === undefined) {
+    throw new Error('The undefined arg type is unsupported by the DAPI.');
+  }
+  if (BN.isBN(value) || typeof value === 'number') {
+    return {
+      type: 'Integer',
+      value,
+    };
+  }
+  if (isUInt160(value)) {
+    return {
+      type: 'Hash160',
+      value,
+    };
+  }
+  if (isUInt256(value)) {
+    return {
+      type: 'Hash256',
+      value,
+    };
+  }
+  if (typeof value === 'string') {
+    return {
+      type: 'String',
+      value,
+    };
+  }
+  if (typeof value === 'boolean') {
+    return {
+      type: 'Boolean',
+      value,
+    };
+  }
+  if (value instanceof Buffer) {
+    return {
+      type: 'ByteArray',
+      value,
+    };
+  }
+  if (Array.isArray(value)) {
+    return {
+      type: 'Array',
+      value: (value as ReadonlyArray<ScriptBuilderParam>).map(paramToArgDataType),
+    };
+  }
+  if (value instanceof Map) {
+    return {
+      type: 'Array',
+      value: Array.from(value).map(([k, v]) => [paramToArgDataType(k), paramToArgDataType(v)]),
+    };
+  }
+  // tslint:disable-next-line strict-type-predicates
+  if (typeof value === 'object') {
+    return {
+      type: 'Array',
+      value: Object.entries(value).map(([k, v]) => [paramToArgDataType(k), paramToArgDataType(v)]),
+    };
+  }
+  throw new Error('Arg type is unsupported by the DAPI.');
+};
+
+export class DapiUserAccountProvider<TProvider extends Provider> extends UserAccountProviderBase
+  implements UserAccountProvider {
+  public readonly currentUserAccount$: Observable<UserAccount | undefined>;
+  public readonly userAccounts$: Observable<ReadonlyArray<UserAccount>>;
+  public readonly networks$: Observable<ReadonlyArray<NetworkType>>;
+  private readonly dapi: Dapi;
+  private readonly initPromise: Promise<void>;
+  private readonly providers: { readonly [network: string]: TProvider };
+  private readonly currentAccountInternal$: BehaviorSubject<UserAccount | undefined>;
+  private readonly userAccountsInternal$: BehaviorSubject<ReadonlyArray<UserAccount>>;
+  private readonly networksInternal$: BehaviorSubject<ReadonlyArray<NetworkType>>;
+  private readonly defaultNetworkInternal$: BehaviorSubject<NetworkType | undefined>;
+
+  public constructor({
+    dapi,
+    providers,
+  }: {
+    readonly dapi: Dapi;
+    readonly providers: { readonly [network: string]: TProvider };
+  }) {
+    super();
+    this.dapi = dapi;
+    this.providers = providers;
```

Comment:
I don't think you need multiple providers here, why can't we just use the same pattern as `LocalUserAccountProvider` which takes a single `Provider`?